File: groups/group1/Lucas.java
import java.util.Scanner;

public class Lucas {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String termsText = scanner.hasNextLine() ? scanner.nextLine() : "";

        if (!hasValidInput(termsText)) {
            return;
        }

        int terms = Integer.parseInt(termsText.trim());
        System.out.println("Lucas sequence:");
        for (int i = 0; i < terms; i++) {
            System.out.print(lucas(i));
            if (i < terms - 1) {
                System.out.print(" ");
            }
        }
        System.out.println();
    }

    static int lucas(int n) {
        if (n == 0) {
            return 2;
        }
        if (n == 1) {
            return 1;
        }
        return lucas(n - 1) + lucas(n - 2);
    }

    static boolean hasValidInput(String termsText) {
        if (!isInteger(termsText)) {
            System.out.println("Invalid input. Enter the number of terms.");
            return false;
        }

        int terms = Integer.parseInt(termsText.trim());
        if (terms <= 0) {
            System.out.println("Invalid input. Number of terms must be positive.");
            return false;
        }

        return true;
    }

    static boolean isInteger(String value) {
        try {
            Integer.parseInt(value.trim());
            return true;
        } catch (NumberFormatException error) {
            return false;
        }
    }
}

File: groups/group1/Lucas.js
const Lucas = (() => {
  const MIN_TERMS = 2;

  function validate(input) {
    const trimmed = String(input ?? "").trim();

    if (trimmed === "") {
      return { ok: false, message: "Please enter a value before calculating." };
    }

    if (!/^\d+$/.test(trimmed)) {
      return { ok: false, message: "Invalid input — please enter a whole number." };
    }

    const value = Number(trimmed);

    if (!Number.isSafeInteger(value)) {
      return {
        ok: false,
        message: "Invalid input — the number of terms is too large to generate safely.",
      };
    }

    if (value <= 0) {
      return { ok: false, message: "Invalid input — please enter a positive integer." };
    }

    if (value <= MIN_TERMS) {
      return {
        ok: false,
        message: `Invalid input — please enter an integer greater than ${MIN_TERMS}.`,
      };
    }

    return { ok: true, message: "", value };
  }

  function compute(n) {
    let previous = 2n;
    let current = 1n;

    for (let i = 0; i < n; i += 1) {
      [previous, current] = [current, previous + current];
    }

    return previous;
  }

  function getSequence(terms) {
    const sequence = [];
    let previous = 2n;
    let current = 1n;

    for (let i = 0; i < terms; i += 1) {
      sequence.push(previous);
      [previous, current] = [current, previous + current];
    }

    return sequence;
  }

  return { MIN_TERMS, validate, compute, getSequence };
})();

function main(inputs) {
  const rawValue = inputs[0];
  const validation = Lucas.validate(rawValue);

  if (!validation.ok) {
    throw new Error(validation.message);
  }

  const terms = validation.value;
  const sequence = Lucas.getSequence(terms).map((value) => value.toString());

  return [
    "Lucas sequence:",
    sequence.join(", "),
    "",
    `Number of terms: ${terms}`,
  ].join("\n");
}
